saleor/checkout/models.py

```diff
@@ -181,8 +181,13 @@ class Checkout(models.Model):
     discount_name = models.CharField(max_length=255, blank=True, null=True)
 
     translated_discount_name = models.CharField(max_length=255, blank=True, null=True)
-    voucher_code = models.CharField(max_length=255, blank=True, null=True)
     gift_cards = models.ManyToManyField(GiftCard, blank=True, related_name="checkouts")
+    voucher_code = models.CharField(max_length=255, blank=True, null=True)
+
+    # The field is to prevent race condition when two different threads are processing
+    # the same checkout with limited usage voucher assigned, both threads increasing the
+    # voucher usage which causing `NotApplicable` error for voucher
```

Comment:
```suggestion
    # The field prevents race condition when two different threads are processing
    # the same checkout with limited usage voucher assigned. Both threads increasing the
    # voucher usage would cause `NotApplicable` error for voucher.
```